pkgs/development/libraries/libbfd/default.nix

```diff
@@ -11,17 +11,20 @@ stdenv.mkDerivation rec {
 
   patches = binutils-raw.bintools.patches ++ [
     ../../tools/misc/binutils/build-components-separately.patch
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/mxe/mxe/e1d4c144ee1994f70f86cf7fd8168fe69bd629c6/src/bfd-1-disable-subdir-doc.patch";
+      sha256 = "0pzb3i74d1r7lhjan376h59a7kirw15j7swwm8pz3zy9lkdqkj6q";
+    })
   ];
 
   # We just want to build libbfd
-  postPatch = ''
+  postAutoreconf = ''
```

Comment:
I'm not sure about this either? We are only building in the `bfd` subdirectory, so shouldn't need to autoreconf the outer stuff. I also don't think the hook is recursive and will properly descend either.

Reply:
Hmm I'll look into this.  FWIW the update hook is recursive: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/update-autotools-gnu-config-scripts.sh .

And unpacking binutils it appears `config.guess` and `config.sub` are at the top-level, not in the `bfd` directory, which is why it's important for the update to happen at the top-level.

(FWIW the problem this fixes is configure complaining it doesn't recognize the platform)

I think what we'd ideally do is this:

* update hook runs at top-level, first
* autoreconf happens, in bfd directory

It appears the hooks are both added to `preConfigurePhases`, with the update happening *after* autoreconf which is unfortunate.  Lemme see..